platform/windows/detect.py

```diff
@@ -100,20 +105,28 @@ def build_res_file(target, source, env):
 
 
 def setup_msvc_manual(env):
+    # FIXME: This is super hacky, and probably obsolete.
+    # Won't work with detect_arch() used for `arch` by default.
+
     """Set up env to use MSVC manually, using VCINSTALLDIR"""
-    if env["bits"] != "default":
+    if env["arch"] != "auto":
         print(
             """
-            Bits argument is not supported for MSVC compilation. Architecture depends on the Native/Cross Compile Tools Prompt/Developer Console
-            (or Visual Studio settings) that is being used to run SCons. As a consequence, bits argument is disabled. Run scons again without bits
-            argument (example: scons p=windows) and SCons will attempt to detect what MSVC compiler will be executed and inform you.
+            Arch argument is not supported for MSVC manual configuration (VCINSTALLDIR configured).
+            Architecture depends on the Native/Cross Compile Tools Prompt/Developer Console (or Visual Studio settings) that is being used to run SCons.
+            As a consequence, the arch argument is disabled. Run scons again without arch argument (example: scons p=windows)
+            and SCons will attempt to detect what MSVC compiler will be executed and inform you.
             """
         )
-        raise SCons.Errors.UserError("Bits argument should not be used when using VCINSTALLDIR")
+        raise SCons.Errors.UserError("Arch argument should not be used when using VCINSTALLDIR")
 
-    # Force bits arg
+    # Force ARCH arg
     # (Actually msys2 mingw can support 64-bit, we could detect that)
-    env["bits"] = "32"
+    # TODO: This is wrong, but not sure what to do about it.
+    # We want to determine the arch and bitness in the SConstruct only.
+    # We can check if it's correct in here, but if it's not, it should
+    # just fail with an error message instead of trying to force it.
+    env["arch"] = "x86_32"
```

Comment:
This should be fine, this part of code is used when you are running scons from a developer console for a specific architecture. Architecture is detected by `detect_visual_c_compiler_version`(which is looking in the `PATH` to get compiler name). But I would move `env["arch"] = "x86_32"` to the `elif compiler_version_str == "x86" ...` branch below.